src/uu/expand/src/expand.rs

```diff
@@ -358,6 +358,14 @@ fn expand_line(
 ) -> std::io::Result<()> {
     use self::CharType::{Backspace, Other, Tab};
 
+    // Fast path: if there are no tabs, backspaces, or carriage returns, just write the buffer
+    if !buf.contains(&b'\t') && !buf.contains(&b'\x08') && (options.uflag || !buf.contains(&b'\r'))
```

Comment:
There is a mismatch between the code and the comment: the `uflag` also seems to play a role in deciding whether the fast path is used.